specification/azurestack/resource-manager/Microsoft.AzureStack/stable/2017-06-01/Product.json

```diff
@@ -143,6 +143,147 @@
+        "x-ms-examples": {
+          "Returns a list of products.": {
+            "$ref": "examples/Product/ListPost.json"
+          }
+        },
+        "tags": [
+          "Product"
+        ],
+        "description": "Returns a list of products.",
+        "operationId": "GetProducts_List",
+        "parameters": [
+          {
+            "$ref": "AzureStack.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "AzureStack.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "AzureStack.json#/parameters/RegistrationNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DeviceConfigurationParameter"
+          },
+          {
+            "$ref": "AzureStack.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ProductList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "AzureStack.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.AzureStack/registrations/{registrationName}/products/{productName}/GetProduct": {
+      "post": {
+        "x-ms-examples": {
+          "Returns the specified product.": {
+            "$ref": "examples/Product/GetPost.json"
+          }
+        },
+        "tags": [
+          "Product"
+        ],
+        "description": "Returns the specified product.",
+        "operationId": "GetProduct",
+        "parameters": [
+          {
+            "$ref": "AzureStack.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "AzureStack.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "AzureStack.json#/parameters/RegistrationNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ProductNameParameter"
+          },
+          {
+            "$ref": "AzureStack.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/DeviceConfigurationParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Product"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "AzureStack.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.AzureStack/registrations/{registrationName}/products/{productName}/uploadProductLog": {
```

Comment:
Sure there is. It's not defined within this file, because system that contains it is not accessible by customers (ACIS operations can access it), and is not described here at all.
Customers and RPs don't need access to that log, it's for our internal use only